src/finalisers/shared/addJsExtension.ts

```diff
@@ -0,0 +1,3 @@
+export default function addJsExtension(name: string): string {
+	return !name.endsWith('.js') ? name + '.js' : name;
```

Comment:
You added a new logic here, was this intended? E.g. if you have an external dependency `dep.amd` you would end up with `dep.amd.js`. And if the user decides to use chunk names without extensions, the extensions would be added anyway in the import, which will probably not work. What is the rationale? Also, it is not tested.

Reply:
Yes, it is. Extension depends on `output.amd.keepExtension`, it described in [updateExtensionForRelativeAmdId](https://github.com/rollup/rollup/pull/4607/commits/b331d2d439d2cd487861509e1612d70fcf46e3da#diff-57bfdcbffed3109dfc5681cfcc5c3e40d75e1b50bd69cf5890e978158b0d7117). 
If `keepExtension` is `true` and module is local dependency (like `./foo`), extension will be added, if it package or external (like `bar` or `lib/baz/quux`) extension will be removed,

It partially tested in [this commit](https://github.com/rollup/rollup/pull/4607/commits/913312a5b6e5f103c922846ca3ceb30c82eb5e9c)

Reply:
> partially

Ah I see, the case that is not tested (and which coverage complains about) is not what I thought but actually the combination relative id + js extension present + keep extension. Maybe you can add this for coverage.

The problem I saw is that
a) there may already be a different extension, so adding the extension `.js` is actually wrong, and
b) there are multiple chunks (e.g. a non-external dynamic import) and the user sets e.g. `output.chunkFileNames: "[name].amd"`. Then "keepExtension" would probably add a wrong extension.

But then again, they should not use `keepExtension` in that case, so I convinced myself that you are right and this is irrelevant (also the way you wrote the docs supports this). Proposal to make this clearer: Change the name to "amd.keepJsExtension" to make clear this is about the presence or absence of ".js"?

Reply:
> Proposal to make this clearer: Change the name to "amd.keepJsExtension" to make clear this is about the presence or absence of ".js"?

I think it should be `amd.addJsExtension` or `amd.foreceJsExtensionForImports`

> a) there may already be a different extension, so adding the extension .js is actually wrong, and

I do not quite understand what is at stake. If you talking about dependencies in `define`, then only js modules will get there. Give me a example and i add tests. 

> b) there are multiple chunks (e.g. a non-external dynamic import) and the user sets e.g. output.chunkFileNames: "[name].amd". Then "keepExtension" would probably add a wrong extension.

No, if `amd.keepExtension: false`. If true, result should be `[name].amd.js`, i add test just in case

Reply:
> amd.foreceJsExtensionForImports

That is an even better name